src/components/UserSelector.tsx

```diff
@@ -1,23 +1,23 @@
-import React, { useContext, useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppSelector } from '../app/hooks';
+import { getUsers } from '../api/users';
+import { useDispatch } from 'react-redux';
+import { setUsers } from '../features/users';
+import { setUser } from '../features/author';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
+export const UserSelector = () => {
   // `users` are loaded from the API, so for the performance reasons
   // we load them once in the `UsersContext` when the `App` is opened
   // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+  const users = useAppSelector(state => state.users);
+  const selectedUser = useAppSelector(state => state.author);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    getUsers().then(array => dispatch(setUsers(array)));
```

Comment:
You are fetching users data every time this component mounts. If the user data doesn't change frequently, consider fetching the data once and storing it in a global state (like Redux) or local storage, and then use that data across your app. Fetching data on every component mount can affect the performance of your application.